src/Eshopworld.WorkerProcess/LeaseAllocator.cs

```diff
@@ -34,83 +35,103 @@ public class LeaseAllocator : ILeaseAllocator
         /// <inheritdoc />
         public async Task<ILease> AllocateLeaseAsync(Guid instanceId)
         {
+            //Lease allocation following design documented here : https://eshopworld.atlassian.net/wiki/spaces/EPLT/pages/2345435290/Worker+Process
+            _telemetry.Publish(new LeaseAcquisitionEvent(instanceId, _options.Value.WorkerType,
+                _options.Value.Priority, $"Lease allocation started for instance {instanceId}"));
+           
             var lease = await _leaseStore.ReadByLeaseTypeAsync(_options.Value.WorkerType).ConfigureAwait(false);
+            if (lease?.LeasedUntil != null && lease.LeasedUntil.Value > ServerDateTime.UtcNow)
+                return null;
+            
+            await _leaseStore.AddLeaseRequestAsync(_options.Value.WorkerType,_options.Value.Priority,instanceId).ConfigureAwait(false);
 
-            if (lease == null)
-            {
-                var leaseResult = await _leaseStore.TryCreateLeaseAsync(_options.Value.WorkerType,
-                    _options.Value.Priority, instanceId).ConfigureAwait(false);
+            // backoff to allow other workers to add their lease request
+            var delay = _allocationDelay.Calculate(_options.Value.Priority, _options.Value.LeaseInterval);
```

Comment:
Do we need this way of calculation? now we expect the delay to be equal for all, I think?

Reply:
yes i agree, but we could change the implementation of IAllocationDelay to provide a fixed delay. This would be also good for unit testing.

Reply:
Agreed to replace with a fixed value provided as parameter in options

Reply:
Removed IAllocationDelay altogether in favor of passing a new option property ElectionDelay